examples/hybrid/sphere/deformation_flow.jl

```diff
@@ -23,412 +25,306 @@ import TerminalLoggers
-    horztopology = Topologies.Topology2D(horzmesh)
-    quad = Spaces.Quadratures.GLL{npoly + 1}()
-    horzspace = Spaces.SpectralElementSpace2D(horztopology, quad)
-
-    hv_center_space =
-        Spaces.ExtrudedFiniteDifferenceSpace(horzspace, vert_center_space)
-    hv_face_space = Spaces.FaceExtrudedFiniteDifferenceSpace(hv_center_space)
-    return (hv_center_space, hv_face_space)
-end
-
-# set up 3D domain
-hv_center_space, hv_face_space = sphere_3D()
-
-# initial conditions
-coords = Fields.coordinate_field(hv_center_space)
-face_coords = Fields.coordinate_field(hv_face_space)
-
-r1(λ, ϕ) = R * acos(sind(ϕ_c) * sind(ϕ) + cosd(ϕ_c) * cosd(ϕ) * cosd(λ - λ_c1))
-r2(λ, ϕ) = R * acos(sind(ϕ_c) * sind(ϕ) + cosd(ϕ_c) * cosd(ϕ) * cosd(λ - λ_c2))
 
+# Reference pressure and density
 p(z) = p_0 * exp(-z / H)
 ρ_ref(z) = p(z) / R_d / T_0
 
-y0 = map(coords) do coord
-    z = coord.z
-    zd = z - z_c
-    λ = coord.long
+# Local wind velocity forcing
+function local_flow(local_geometry, t)
+    coord = local_geometry.coordinates
     ϕ = coord.lat
-    rd1 = r1(λ, ϕ)
-    rd2 = r2(λ, ϕ)
-
-    d1 = min(1, (rd1 / R_t)^2 + (zd / Z_t)^2)
-    d2 = min(1, (rd2 / R_t)^2 + (zd / Z_t)^2)
-
-    q1 = 1 / 2.0 * (1 + cos(pi * d1)) + 1 / 2.0 * (1 + cos(pi * d2))
-    q2 = 0.9 - 0.8 * q1^2
-    q3 = 0.0
-    if d1 < 0.5 || d2 < 0.5
-        q3 = 1.0
-    else
-        q3 = 0.1
-    end
+    λ = coord.long
+    z = coord.z
 
-    if (z > z_c) && (ϕ > ϕ_c - rad2deg(1 / 8)) && (ϕ < ϕ_c + rad2deg(1 / 8))
-        q3 = 0.1
-    end
-    q4 = 1 - 3 / 10 * (q1 + q2 + q3)
+    k = 10 * R / τ
+    λp = λ - FT(360) * t / τ
 
-    ρq1 = ρ_ref(z) * q1
-    ρq2 = ρ_ref(z) * q2
-    ρq3 = ρ_ref(z) * q3
-    ρq4 = ρ_ref(z) * q4
+    ua =
+        k * sind(λp)^2 * sind(2 * ϕ) * cos(FT(π) * t / τ) +
+        2 * FT(π) * R / τ * cosd(ϕ)
+    ud =
+        ω_0 * R / b / p_top *
+        cosd(λp) *
+        cosd(ϕ)^2 *
+        cos(2 * FT(π) * t / τ) *
+        (-exp((p(z) - p_0) / b / p_top) + exp((p_top - p(z)) / b / p_top))
+    uu = ua + ud
 
-    return (
-        ρ = ρ_ref(z),
-        ρq1 = ρq1,
-        ρq2 = ρq2,
-        ρq3 = ρq3,
-        ρq4 = ρq4,
-        ρq1_td = ρq1,
-        ρq2_td = ρq2,
-        ρq3_td = ρq3,
-        ρq4_td = ρq4,
-    )
+    uv = k * sind(2 * λp) * cosd(ϕ) * cos(FT(π) * t / τ)
+
+    sp =
+        1 + exp((p_top - p_0) / b / p_top) - exp((p(z) - p_0) / b / p_top) -
+        exp((p_top - p(z)) / b / p_top)
+    ω = ω_0 * sind(λp) * cosd(ϕ) * cos(2 * FT(π) * t / τ) * sp
+    uw = -ω / ρ_ref(z) / grav
+
+    Geometry.Covariant123Vector(Geometry.UVWVector(uu, uv, uw), local_geometry)
 end
 
-y0 = Fields.FieldVector(
-    ρ = y0.ρ,
-    ρq1 = y0.ρq1,
-    ρq2 = y0.ρq2,
-    ρq3 = y0.ρq3,
-    ρq4 = y0.ρq4,
-    ρq1_td = y0.ρq1,
-    ρq2_td = y0.ρq2,
-    ρq3_td = y0.ρq3,
-    ρq4_td = y0.ρq4,
-)
+# "Tendency increment", which computes Y2 .+= dt * ∂Y/∂t(Y1, t)
+function increment!(Y2, Y1, cache, t, dt, beta)
```

Comment:
Thank you very much Dennis for your work in unifying and improving this. Would you please consider renaming Y2 something along the lines of `Ystar` (as an intermediate stage solution) and `Y1` to `Y` only? (And similarly `ρ2` and `ρq2` as `ρstar` and `ρqstar`, respectively). I am afraid this numeral convention chosen can become easily confusing with the fact that originally `ρq2` was referring to tracer number 2, and so on.

Reply:
How about `Y⁺`, where the plus indicates that the variable is being incremented?